In [183]:
import pandas as pd
import numpy as np
import requests
import json
import time
from time import sleep
import random
import spotipy
from random import randint
import glob
import re
from spotipy.oauth2 import SpotifyClientCredentials
client_id = ""
client_secret = ""
client_credentials_manager = SpotifyClientCredentials(client_id, client_secret)
sp = spotipy.Spotify(client_credentials_manager=client_credentials_manager)

In [2]:
cd C:\Users\Dubun\IronHack\Other\jupyter_notebook

C:\Users\Dubun\IronHack\Other\jupyter_notebook


# NARROWING EXISTING DATABASE FROM KAGGLE

In [10]:
data=pd.read_csv('Artist-Genres-URIs.csv')
data
#source https://www.kaggle.com/toluwafayemi/66m-spotify-artists

Unnamed: 0                           name genres  \
0                 0  Lisztes Jenő Cimbalom Project     []   
1                 1                         Stewie     []   
2                 2                          G.O'd     []   
3                 3         Zico Aka Lost Identity     []   
4                 4                  Suburban Drag     []   
...             ...                            ...    ...   
6690297     6690297                 Gifted Elsueno     []   
6690298     6690298                  Dylan Peirens     []   
6690299     6690299                 Jenny Meloncon     []   
6690300     6690300                      Deezy Moe     []   
6690301     6690301                 Richard Faltin     []   

                                   spotify_uri  
0        spotify:artist:0NxUJYwYtXrUHNNwQznLwL  
1        spotify:artist:0VLFOxJlBB2LYNttSejrEv  
2        spotify:artist:0ijndZDbEsSTgDS8qyw6fI  
3        spotify:artist:0xB16BIgyn5Uy5TtNGVEEw  
4        spotify:artist:3FxfEhNrYCMiZl6f9ajrbE  
...                                        ...  
6690297  spotify:artist:4nAIBPWDyqW6lG1jYvL8hB  
6690298  spotify:artist:5EcPQyxRcBcig5jfX1RFXb  
6690299  spotify:artist:5J0SmsrilOV6pwFSXRllNd  
6690300  spotify:artist:5qd5nZ0JFI8ZGOENZFEMDR  
6690301  spotify:artist:79AONyJ23aC3fk8xCTu3Oz  

[6690302 rows x 4 columns]

In [14]:
french1 = data[data['genres'].str.contains("french")]
french2 = data[data['genres'].str.contains("francais")]
df_fr = pd.concat([french1,french2])
df_fr = df_fr.drop_duplicates(subset=['spotify_uri'])
df_fr

Unnamed: 0               name  \
306             306            Pejmaxx   
647             647      Junior Tshaka   
688             688          Psygnosis   
1244           1244  La Maison Tellier   
3420           3420      Julien Brocal   
...             ...                ...   
6609352     6609352            Rahmann   
6656442     6656442         Schérazade   
6677949     6677949            Perkins   
6689568     6689568           CASANOVA   
6690031     6690031     Marie Espinosa   

                                                genres  \
306                     ['french hip hop', 'rap inde']   
647                  ['french reggae', 'swiss reggae']   
688                                   ['french metal']   
1244     ['french rock', 'nouvelle chanson francaise']   
3420                        ['french classical piano']   
...                                                ...   
6609352          ['rock progressif francais', 'zeuhl']   
6656442                       ['electro-pop francais']   
6677949                   ['rock progressif francais']   
6689568            ['eurovision', 'variete francaise']   
6690031                 ['nouvelle chanson francaise']   

                                   spotify_uri  
306      spotify:artist:6gcXOa9ncLivvQHMl1sJU1  
647      spotify:artist:0JncPZTYhYvUG5c7jr6QY8  
688      spotify:artist:5saAU9PdyTVAxsx15SIzO3  
1244     spotify:artist:41EfC0kQmOWfbW2FBGttze  
3420     spotify:artist:5xFAEWAcB0A5e7qk3UO2pc  
...                                        ...  
6609352  spotify:artist:5cWE8KSWcqSCtJ8RVOiOox  
6656442  spotify:artist:3ma9rtafrCvASmUy7WHsUh  
6677949  spotify:artist:2GB1UbK4Lal9GEuEivT1tl  
6689568  spotify:artist:5LFX3E1nLxOMLWOHSFwbG0  
6690031  spotify:artist:2bS6bVJcPzrgqmuxeJsljn  

[4789 rows x 4 columns]

I am interested in the french music market. I think running a machine learning algorithm on all countries would not make sense since accuracy scores would be low. Music consumption is different from one country/region/culture to another.

In [33]:
rap = df_fr[df_fr['genres'].str.contains("rap")]
hip_hop = df_fr[df_fr['genres'].str.contains("hip hop")]
df_hip_hop = pd.concat([hip_hop,rap])
df_hip_hop = df_hip_hop.drop_duplicates(subset=['spotify_uri'])
df_hip_hop['genre']='hip_hop'
df_hip_hop

Unnamed: 0        name  \
306             306     Pejmaxx   
6898           6898     Nefaste   
22607         22607         UZI   
66276         66276     Lorenzo   
74060         74060       Nusky   
...             ...         ...   
6541382     6541382  Dj Crazy B   
6556626     6556626      Boumso   
6562648     6562648      Eliott   
6576965     6576965        Dika   
6607614     6607614       Dasko   

                                                    genres  \
306                         ['french hip hop', 'rap inde']   
6898                        ['french hip hop', 'rap inde']   
22607    ['francoton', 'french hip hop', 'pop urbaine',...   
66276    ['francoton', 'french hip hop', 'pop urbaine',...   
74060                                   ['french hip hop']   
...                                                    ...   
6541382                        ['old school rap francais']   
6556626                    ['pop urbaine', 'rap francais']   
6562648                                   ['rap francais']   
6576965                  ['rap francais', 'rap marseille']   
6607614                                   ['rap francais']   

                                   spotify_uri    genre  
306      spotify:artist:6gcXOa9ncLivvQHMl1sJU1  hip_hop  
6898     spotify:artist:0H1xpJStOZvjAvCihf95aY  hip_hop  
22607    spotify:artist:5U05AU34EcZ8pfohg2dicI  hip_hop  
66276    spotify:artist:62zLrboqOTPyozwi721CVy  hip_hop  
74060    spotify:artist:6NBA1GBrRG7fxTPDjKwFxm  hip_hop  
...                                        ...      ...  
6541382  spotify:artist:0t6f8MpJvTEuTuHlsK2Fls  hip_hop  
6556626  spotify:artist:6C91nXJJkhPEoc5thhKiNn  hip_hop  
6562648  spotify:artist:6wBLSFOSNFgdUme2WHGCWv  hip_hop  
6576965  spotify:artist:1rBnd8KxDndiiiIyjOsFtw  hip_hop  
6607614  spotify:artist:3nMGcr43bHLpmyIjLFyTaj  hip_hop  

[666 rows x 5 columns]

From the french music market, I will explore main genres.

I am bundling rap and hip hop into the rap category.

In [34]:
pop = df_fr[df_fr['genres'].str.contains("pop")]
variete = df_fr[df_fr['genres'].str.contains("variete")]
df_pop = pd.concat([pop,variete])
df_pop = df_pop.drop_duplicates(subset=['spotify_uri'])
df_pop['genre']='pop'
df_pop

Unnamed: 0                name  \
22607         22607                 UZI   
23904         23904  Electric Bazar Cie   
33265         33265            Grégoire   
43636         43636       Colonel Reyel   
54071         54071   La Bande à Basile   
...             ...                 ...   
5777138     5777138          La Deryves   
6119059     6119059           Andriamad   
6361337     6361337               Fab'M   
6449892     6449892                 KIZ   
6689568     6689568            CASANOVA   

                                                    genres  \
22607    ['francoton', 'french hip hop', 'pop urbaine',...   
23904                                  ['french folk pop']   
33265               ['chanson', 'francoton', 'french pop']   
43636           ['francoton', 'french pop', 'pop urbaine']   
54071          ['chanson paillarde', 'classic french pop']   
...                                                    ...   
5777138                              ['variete francaise']   
6119059                ['eurovision', 'variete francaise']   
6361337                              ['variete francaise']   
6449892                              ['variete francaise']   
6689568                ['eurovision', 'variete francaise']   

                                   spotify_uri genre  
22607    spotify:artist:5U05AU34EcZ8pfohg2dicI   pop  
23904    spotify:artist:3YM53hfUFLh3XcpKOQNZcf   pop  
33265    spotify:artist:1WycOcJCIRAH5xyzHXtp1E   pop  
43636    spotify:artist:5gNYKVooIJ94WPmhZXbKMz   pop  
54071    spotify:artist:3Zk4rZqHG6Bl5WwrekbtNK   pop  
...                                        ...   ...  
5777138  spotify:artist:2Xy5B4VG3Q1w10EODepWUk   pop  
6119059  spotify:artist:48TEPWuSU7KRr5sFpafo5q   pop  
6361337  spotify:artist:3bF6ab2QpnAww6OZ4qZM8E   pop  
6449892  spotify:artist:2y7twK2fqUipb5Ef2hNrdO   pop  
6689568  spotify:artist:5LFX3E1nLxOMLWOHSFwbG0   pop  

[1146 rows x 5 columns]

I am above bundling pop and "variété" into pop music. Variété is a french category very close to pop music

In [35]:
electro = df_fr[df_fr['genres'].str.contains("electro")]
house = df_fr[df_fr['genres'].str.contains("house")]
df_electro = pd.concat([electro,house])
df_electro = df_electro.drop_duplicates(subset=['spotify_uri'])
df_electro['genre']='electro'
df_electro

Unnamed: 0                  name  \
37721         37721      Philippe Laurent   
54244         54244              Moriarty   
107598       107598            Venus Gang   
232727       232727              A. Kalma   
307739       307739               Vitalic   
...             ...                   ...   
6479515     6479515              Goldroom   
6519650     6519650                Fakear   
6529312     6529312    Olivier Giacomotto   
6620869     6620869  Vijay & Sofia Zlatko   
6644620     6644620                Aslove   

                                                    genres  \
37721        ['minimal wave', 'vintage french electronic']   
54244    ['chanson', 'electro-pop francais', 'french in...   
107598   ['rock progressif francais', 'vintage french e...   
232727      ['library music', 'vintage french electronic']   
307739   ['alternative dance', 'dance-punk', 'electro h...   
...                                                    ...   
6479515  ['alternative dance', 'chillwave', 'filter hou...   
6519650  ['french indie pop', 'french indietronica', 'n...   
6529312                    ['french techno', 'tech house']   
6620869  ['deep house', 'deep tropical house', 'new fre...   
6644620        ['deep tropical house', 'new french touch']   

                                   spotify_uri    genre  
37721    spotify:artist:6d57Upldy2amVudfPz9gH8  electro  
54244    spotify:artist:0qG2Tf6fJaCuWKGz1IitKa  electro  
107598   spotify:artist:1PzMmjffwQICWezh3DVWa5  electro  
232727   spotify:artist:3C40R1ioWH5ki0nXr30gLZ  electro  
307739   spotify:artist:4M84umUNRbZy1mJleyyRM9  electro  
...                                        ...      ...  
6479515  spotify:artist:4eZebkMFU3xelF8mbZYXyl  electro  
6519650  spotify:artist:4eFImh8D3F15dtZk0JQlpT  electro  
6529312  spotify:artist:5tIMc0MdfB2OV6sULOmeao  electro  
6620869  spotify:artist:23BtKJexFLn0Ri6s7BEC4d  electro  
6644620  spotify:artist:1EnscKRqKyq19bQ2bl642c  electro  

[322 rows x 5 columns]

I am above bundling electro and house as electro music. 

In [36]:
df_rock = df_fr[df_fr['genres'].str.contains("rock")]
df_rock['genre']='rock'
df_rock

<ipython-input-36-579f2912f352>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_rock['genre']='rock'


Unnamed: 0                 name  \
1244           1244    La Maison Tellier   
6932           6932             Anathème   
54244         54244             Moriarty   
57379         57379       100% Chevalier   
60968         60968             La Femme   
...             ...                  ...   
6521397     6521397  Jean-Pierre Alarcen   
6539131     6539131            Watertank   
6572345     6572345              Ame Son   
6609352     6609352              Rahmann   
6677949     6677949              Perkins   

                                                    genres  \
1244         ['french rock', 'nouvelle chanson francaise']   
6932                                  ['french post-rock']   
54244    ['chanson', 'electro-pop francais', 'french in...   
57379                                 ['french post-rock']   
60968    ['french indie pop', 'french indietronica', 'f...   
...                                                    ...   
6521397                       ['rock progressif francais']   
6539131                       ['rock alternatif francais']   
6572345                       ['rock progressif francais']   
6609352              ['rock progressif francais', 'zeuhl']   
6677949                       ['rock progressif francais']   

                                   spotify_uri genre  
1244     spotify:artist:41EfC0kQmOWfbW2FBGttze  rock  
6932     spotify:artist:5XrlxolRDQ4RLX75nkVFEk  rock  
54244    spotify:artist:0qG2Tf6fJaCuWKGz1IitKa  rock  
57379    spotify:artist:4cosRYKPuU1M79bSbZcK3D  rock  
60968    spotify:artist:5VTWoYYizcOY3uIKnxeCGI  rock  
...                                        ...   ...  
6521397  spotify:artist:7DZEj1KbfJLIlY1qq1gSMp  rock  
6539131  spotify:artist:2N3PHaigRY80nInbgjjbPC  rock  
6572345  spotify:artist:46Ek6Eoi6zFtJSl2RkHfOI  rock  
6609352  spotify:artist:5cWE8KSWcqSCtJ8RVOiOox  rock  
6677949  spotify:artist:2GB1UbK4Lal9GEuEivT1tl  rock  

[614 rows x 5 columns]

The last category I will be focusing is rock.

In [51]:
df = pd.concat([df_rock,df_hip_hop, df_electro, df_pop])
df = df.drop_duplicates(subset=['spotify_uri'])
df['spotify_uri'] = df['spotify_uri'].str.replace(r'spotify:artist:', '')
df = df.rename(columns={'name': 'artist_name', 'genre': 'artist_genre', 'spotify_uri': 'artist_id'})
df = df.drop(['Unnamed: 0','genres'], axis = 1)
df

artist_name               artist_id artist_genre
1244     La Maison Tellier  41EfC0kQmOWfbW2FBGttze         rock
6932              Anathème  5XrlxolRDQ4RLX75nkVFEk         rock
54244             Moriarty  0qG2Tf6fJaCuWKGz1IitKa         rock
57379       100% Chevalier  4cosRYKPuU1M79bSbZcK3D         rock
60968             La Femme  5VTWoYYizcOY3uIKnxeCGI         rock
...                    ...                     ...          ...
5777138         La Deryves  2Xy5B4VG3Q1w10EODepWUk          pop
6119059          Andriamad  48TEPWuSU7KRr5sFpafo5q          pop
6361337              Fab'M  3bF6ab2QpnAww6OZ4qZM8E          pop
6449892                KIZ  2y7twK2fqUipb5Ef2hNrdO          pop
6689568           CASANOVA  5LFX3E1nLxOMLWOHSFwbG0          pop

[2245 rows x 3 columns]

I am here assembling the four categories into one dataframe.

# SPOTIPY TO GET TOP10 TRACKS BY ARTIST AND AUDIO FEAT

In [127]:
artist_ids = df['artist_id'].tolist()
artist_genres = df['artist_genre'].tolist()
artist_names = df['artist_name'].tolist()

In [128]:
track_id = []
artist_id = []
artist_nme = []
artist_gnre = []
for i in range(0, len(artist_ids)):
    print(i)
    tracks = sp.artist_top_tracks(artist_ids[i])
    for track in tracks['tracks']:
        artist_id.append(artist_ids[i])
        artist_nme.append(artist_names[i])
        artist_gnre.append(artist_genres[i])
        track_id.append(track['id'])
df1 = pd.DataFrame({'track_id' : track_id, 'artist_id': artist_id, 'artist_name': artist_nme, 'artist_genre': artist_gnre})
df1

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199
200
201
202
203
204
205
206
207
208
209
210
211
212
213
214
215
216
217
218
219
220
221
222
223
224
225
226
227
228
229
230
231
232
233
234
235
236
237
238
239
240
241
242
243
244
245
246
247
248
249
250
251
252
253
254
255
256
257
258
259
260
261
262
263
264
265
266
267
268
269
270
271
272
273
274
275
276
27

1864
1865
1866
1867
1868
1869
1870
1871
1872
1873
1874
1875
1876
1877
1878
1879
1880
1881
1882
1883
1884
1885
1886
1887
1888
1889
1890
1891
1892
1893
1894
1895
1896
1897
1898
1899
1900
1901
1902
1903
1904
1905
1906
1907
1908
1909
1910
1911
1912
1913
1914
1915
1916
1917
1918
1919
1920
1921
1922
1923
1924
1925
1926
1927
1928
1929
1930
1931
1932
1933
1934
1935
1936
1937
1938
1939
1940
1941
1942
1943
1944
1945
1946
1947
1948
1949
1950
1951
1952
1953
1954
1955
1956
1957
1958
1959
1960
1961
1962
1963
1964
1965
1966
1967
1968
1969
1970
1971
1972
1973
1974
1975
1976
1977
1978
1979
1980
1981
1982
1983
1984
1985
1986
1987
1988
1989
1990
1991
1992
1993
1994
1995
1996
1997
1998
1999
2000
2001
2002
2003
2004
2005
2006
2007
2008
2009
2010
2011
2012
2013
2014
2015
2016
2017
2018
2019
2020
2021
2022
2023
2024
2025
2026
2027
2028
2029
2030
2031
2032
2033
2034
2035
2036
2037
2038
2039
2040
2041
2042
2043
2044
2045
2046
2047
2048
2049
2050
2051
2052
2053
2054
2055
2056
2057
2058
2059
2060
2061
2062
2063


track_id               artist_id        artist_name  \
0      3RvVFTMSlEXJjtoLrcDflF  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
1      5w8bF2RGgf9PkXEoEl8SSN  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
2      5u9zU1N1urKWaULrbEhMpr  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
3      1TC9cYi9ZlEAhpoyd5sYn4  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
4      2YiGsf5CLUtgty23h1RNOQ  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
...                       ...                     ...                ...   
21111  4kP1f09mjeK56QEykWC7Sk  2y7twK2fqUipb5Ef2hNrdO                KIZ   
21112  0LA1EZMkLVyafay8DQjS21  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21113  6JYM6Dz5szNhAIHlAd37Cb  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21114  6frPNZ8woC8jH3rgjRRSUL  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21115  7xR1eN0b55dXhf4eyIXTZl  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   

      artist_genre  
0             rock  
1             rock  
2             rock  
3             rock  
4             rock  
...            ...  
21111          pop  
21112          pop  
21113          pop  
21114          pop  
21115          pop  

[21116 rows x 4 columns]

Using spotipy API to retrieve top 10 tracks for every french artist being into one of the four main categories of focus.

In [129]:
df1

track_id               artist_id        artist_name  \
0      3RvVFTMSlEXJjtoLrcDflF  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
1      5w8bF2RGgf9PkXEoEl8SSN  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
2      5u9zU1N1urKWaULrbEhMpr  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
3      1TC9cYi9ZlEAhpoyd5sYn4  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
4      2YiGsf5CLUtgty23h1RNOQ  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
...                       ...                     ...                ...   
21111  4kP1f09mjeK56QEykWC7Sk  2y7twK2fqUipb5Ef2hNrdO                KIZ   
21112  0LA1EZMkLVyafay8DQjS21  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21113  6JYM6Dz5szNhAIHlAd37Cb  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21114  6frPNZ8woC8jH3rgjRRSUL  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21115  7xR1eN0b55dXhf4eyIXTZl  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   

      artist_genre  
0             rock  
1             rock  
2             rock  
3             rock  
4             rock  
...            ...  
21111          pop  
21112          pop  
21113          pop  
21114          pop  
21115          pop  

[21116 rows x 4 columns]

In [131]:
track_ids = df1['track_id'].tolist()
len(track_ids)

21116

In [133]:
audio_feat_fetched=[]
for i in range(0, len(track_ids), 100):
    print(i)
    audio_features_fetched = sp.audio_features(df1['track_id'][i:i+100])
    audio_features_fetched = [x for x in audio_features_fetched if x is not None]
    if len(audio_features_fetched) > 0:
        audio_feat_fetched+=audio_features_fetched
    else: 
        pass


0
100
200
300
400
500
600
700
800
900
1000
1100
1200
1300
1400
1500
1600
1700
1800
1900
2000
2100
2200
2300
2400
2500
2600
2700
2800
2900
3000
3100
3200
3300
3400
3500
3600
3700
3800
3900
4000
4100
4200
4300
4400
4500
4600
4700
4800
4900
5000
5100
5200
5300
5400
5500
5600
5700
5800
5900
6000
6100
6200
6300
6400
6500
6600
6700
6800
6900
7000
7100
7200
7300
7400
7500
7600
7700
7800
7900
8000
8100
8200
8300
8400
8500
8600
8700
8800
8900
9000
9100
9200
9300
9400
9500
9600
9700
9800
9900
10000
10100
10200
10300
10400
10500
10600
10700
10800
10900
11000
11100
11200
11300
11400
11500
11600
11700
11800
11900
12000
12100
12200
12300
12400
12500
12600
12700
12800
12900
13000
13100
13200
13300
13400
13500
13600
13700
13800
13900
14000
14100
14200
14300
14400
14500
14600
14700
14800
14900
15000
15100
15200
15300
15400
15500
15600
15700
15800
15900
16000
16100
16200
16300
16400
16500
16600
16700
16800
16900
17000
17100
17200
17300
17400
17500
17600
17700
17800
17900
18000
18100
18200
18300
18400
18

The above query enables to get the audio features of each of the track from the previous query using spotipy.

In [134]:
df_fetched=pd.DataFrame(audio_feat_fetched)
df_fetched = df_fetched.drop_duplicates(subset=['id'])
df_fetched

danceability  energy  key  loudness  mode  speechiness  acousticness  \
0             0.499   0.362    7   -16.331     1       0.0378      0.186000   
1             0.398   0.685    7    -9.098     0       0.0417      0.417000   
2             0.714   0.928    0    -7.118     1       0.0385      0.034300   
3             0.519   0.728    4    -6.660     0       0.0290      0.103000   
4             0.574   0.635    3    -6.683     1       0.0242      0.000765   
...             ...     ...  ...       ...   ...          ...           ...   
21109         0.762   0.451    5    -9.501     0       0.0848      0.257000   
21110         0.709   0.724    3    -4.928     1       0.0608      0.316000   
21111         0.429   0.787    5    -6.510     0       0.2840      0.075100   
21112         0.724   0.686    8    -5.040     0       0.0660      0.155000   
21113         0.620   0.666    1    -5.355     0       0.0762      0.392000   

       instrumentalness  liveness  valence    tempo            type  \
0              0.175000    0.1190    0.354  177.691  audio_features   
1              0.005580    0.0787    0.376  109.994  audio_features   
2              0.746000    0.1370    0.487  120.007  audio_features   
3              0.000518    0.0652    0.321   86.026  audio_features   
4              0.001020    0.3640    0.521   99.994  audio_features   
...                 ...       ...      ...      ...             ...   
21109          0.000000    0.0984    0.874  129.918  audio_features   
21110          0.000000    0.0623    0.495  109.079  audio_features   
21111          0.000000    0.0879    0.563   60.031  audio_features   
21112          0.000000    0.0544    0.750  119.031  audio_features   
21113          0.000000    0.0704    0.455  103.035  audio_features   

                           id                                   uri  \
0      3RvVFTMSlEXJjtoLrcDflF  spotify:track:3RvVFTMSlEXJjtoLrcDflF   
1      5w8bF2RGgf9PkXEoEl8SSN  spotify:track:5w8bF2RGgf9PkXEoEl8SSN   
2      5u9zU1N1urKWaULrbEhMpr  spotify:track:5u9zU1N1urKWaULrbEhMpr   
3      1TC9cYi9ZlEAhpoyd5sYn4  spotify:track:1TC9cYi9ZlEAhpoyd5sYn4   
4      2YiGsf5CLUtgty23h1RNOQ  spotify:track:2YiGsf5CLUtgty23h1RNOQ   
...                       ...                                   ...   
21109  4kP1f09mjeK56QEykWC7Sk  spotify:track:4kP1f09mjeK56QEykWC7Sk   
21110  0LA1EZMkLVyafay8DQjS21  spotify:track:0LA1EZMkLVyafay8DQjS21   
21111  6JYM6Dz5szNhAIHlAd37Cb  spotify:track:6JYM6Dz5szNhAIHlAd37Cb   
21112  6frPNZ8woC8jH3rgjRRSUL  spotify:track:6frPNZ8woC8jH3rgjRRSUL   
21113  7xR1eN0b55dXhf4eyIXTZl  spotify:track:7xR1eN0b55dXhf4eyIXTZl   

                                              track_href  \
0      https://api.spotify.com/v1/tracks/3RvVFTMSlEXJ...   
1      https://api.spotify.com/v1/tracks/5w8bF2RGgf9P...   
2      https://api.spotify.com/v1/tracks/5u9zU1N1urKW...   
3      https://api.spotify.com/v1/tracks/1TC9cYi9ZlEA...   
4      https://api.spotify.com/v1/tracks/2YiGsf5CLUtg...   
...                                                  ...   
21109  https://api.spotify.com/v1/tracks/4kP1f09mjeK5...   
21110  https://api.spotify.com/v1/tracks/0LA1EZMkLVya...   
21111  https://api.spotify.com/v1/tracks/6JYM6Dz5szNh...   
21112  https://api.spotify.com/v1/tracks/6frPNZ8woC8j...   
21113  https://api.spotify.com/v1/tracks/7xR1eN0b55dX...   

                                            analysis_url  duration_ms  \
0      https://api.spotify.com/v1/audio-analysis/3RvV...       234507   
1      https://api.spotify.com/v1/audio-analysis/5w8b...       185334   
2      https://api.spotify.com/v1/audio-analysis/5u9z...       216000   
3      https://api.spotify.com/v1/audio-analysis/1TC9...       226640   
4      https://api.spotify.com/v1/audio-analysis/2YiG...       246467   
...                                                  ...          ...   
21109  https://api.spotify.com/v1/audio-analysis/4kP1...       188760   
21110  https://api.spotify.com/v1/audio-analysi

droping duplicates in order to have track_id as a unique identifier.

In [135]:
df2 = df1.merge(df_fetched, left_on = 'track_id', right_on = 'id')
df2

track_id               artist_id        artist_name  \
0      3RvVFTMSlEXJjtoLrcDflF  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
1      5w8bF2RGgf9PkXEoEl8SSN  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
2      5u9zU1N1urKWaULrbEhMpr  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
3      1TC9cYi9ZlEAhpoyd5sYn4  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
4      2YiGsf5CLUtgty23h1RNOQ  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
...                       ...                     ...                ...   
21109  4kP1f09mjeK56QEykWC7Sk  2y7twK2fqUipb5Ef2hNrdO                KIZ   
21110  0LA1EZMkLVyafay8DQjS21  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21111  6JYM6Dz5szNhAIHlAd37Cb  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21112  6frPNZ8woC8jH3rgjRRSUL  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21113  7xR1eN0b55dXhf4eyIXTZl  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   

      artist_genre  danceability  energy  key  loudness  mode  speechiness  \
0             rock         0.499   0.362    7   -16.331     1       0.0378   
1             rock         0.398   0.685    7    -9.098     0       0.0417   
2             rock         0.714   0.928    0    -7.118     1       0.0385   
3             rock         0.519   0.728    4    -6.660     0       0.0290   
4             rock         0.574   0.635    3    -6.683     1       0.0242   
...            ...           ...     ...  ...       ...   ...          ...   
21109          pop         0.762   0.451    5    -9.501     0       0.0848   
21110          pop         0.709   0.724    3    -4.928     1       0.0608   
21111          pop         0.429   0.787    5    -6.510     0       0.2840   
21112          pop         0.724   0.686    8    -5.040     0       0.0660   
21113          pop         0.620   0.666    1    -5.355     0       0.0762   

       ...  liveness  valence    tempo            type  \
0      ...    0.1190    0.354  177.691  audio_features   
1      ...    0.0787    0.376  109.994  audio_features   
2      ...    0.1370    0.487  120.007  audio_features   
3      ...    0.0652    0.321   86.026  audio_features   
4      ...    0.3640    0.521   99.994  audio_features   
...    ...       ...      ...      ...             ...   
21109  ...    0.0984    0.874  129.918  audio_features   
21110  ...    0.0623    0.495  109.079  audio_features   
21111  ...    0.0879    0.563   60.031  audio_features   
21112  ...    0.0544    0.750  119.031  audio_features   
21113  ...    0.0704    0.455  103.035  audio_features   

                           id                                   uri  \
0      3RvVFTMSlEXJjtoLrcDflF  spotify:track:3RvVFTMSlEXJjtoLrcDflF   
1      5w8bF2RGgf9PkXEoEl8SSN  spotify:track:5w8bF2RGgf9PkXEoEl8SSN   
2      5u9zU1N1urKWaULrbEhMpr  spotify:track:5u9zU1N1urKWaULrbEhMpr   
3      1TC9cYi9ZlEAhpoyd5sYn4  spotify:track:1TC9cYi9ZlEAhpoyd5sYn4   
4      2YiGsf5CLUtgty23h1RNOQ  spotify:track:2YiGsf5CLUtgty23h1RNOQ   
...                       ...                                   ...   
21109  4kP1f09mjeK56QEykWC7Sk  spotify:track:4kP1f09mjeK56QEykWC7Sk   
21110  0LA1EZMkLVyafay8DQjS21  spotify:track:0LA1EZMkLVyafay8DQjS21   
21111  6JYM6Dz5szNhAIHlAd37Cb  spotify:track:6JYM6Dz5szNhAIHlAd37Cb   
21112  6frPNZ8woC8jH3rgjRRSUL  spotify:track:6frPNZ8woC8jH3rgjRRSUL   
21113  7xR1eN0b55dXhf4eyIXTZl  spotify:track:7xR1eN0b55dXhf4eyIXTZl   

                                              track_href  \
0      https://api.spotify.com/v1/tracks/3RvVFTMSlEXJ...   
1      https://api.spotify.com/v1/tracks/5w8bF2RGgf9P...   
2      https://api.spotify.com/v1/tracks/5u9zU1N1urKW...   
3      https://api.spotify.com/v1/tracks/1TC9cYi9ZlEA...   
4      https://api.spotify.com/v1/tracks/2YiGsf5CLUtg...   
...                                                  ...   
21109  https://api.spotify.com/v1/tracks/4kP1f09mjeK5...   
21110  https://api.spotify.com/v1/tracks/0LA1EZMkLVya...   
21111  https://api.spotify.com/v1/tracks/6JYM6Dz5szNh...   
21112  https://api.sp

In [136]:
df2 = df2.drop(['type','id','uri','track_href','analysis_url'], axis = 1)
df2

track_id               artist_id        artist_name  \
0      3RvVFTMSlEXJjtoLrcDflF  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
1      5w8bF2RGgf9PkXEoEl8SSN  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
2      5u9zU1N1urKWaULrbEhMpr  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
3      1TC9cYi9ZlEAhpoyd5sYn4  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
4      2YiGsf5CLUtgty23h1RNOQ  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
...                       ...                     ...                ...   
21109  4kP1f09mjeK56QEykWC7Sk  2y7twK2fqUipb5Ef2hNrdO                KIZ   
21110  0LA1EZMkLVyafay8DQjS21  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21111  6JYM6Dz5szNhAIHlAd37Cb  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21112  6frPNZ8woC8jH3rgjRRSUL  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21113  7xR1eN0b55dXhf4eyIXTZl  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   

      artist_genre  danceability  energy  key  loudness  mode  speechiness  \
0             rock         0.499   0.362    7   -16.331     1       0.0378   
1             rock         0.398   0.685    7    -9.098     0       0.0417   
2             rock         0.714   0.928    0    -7.118     1       0.0385   
3             rock         0.519   0.728    4    -6.660     0       0.0290   
4             rock         0.574   0.635    3    -6.683     1       0.0242   
...            ...           ...     ...  ...       ...   ...          ...   
21109          pop         0.762   0.451    5    -9.501     0       0.0848   
21110          pop         0.709   0.724    3    -4.928     1       0.0608   
21111          pop         0.429   0.787    5    -6.510     0       0.2840   
21112          pop         0.724   0.686    8    -5.040     0       0.0660   
21113          pop         0.620   0.666    1    -5.355     0       0.0762   

       acousticness  instrumentalness  liveness  valence    tempo  \
0          0.186000          0.175000    0.1190    0.354  177.691   
1          0.417000          0.005580    0.0787    0.376  109.994   
2          0.034300          0.746000    0.1370    0.487  120.007   
3          0.103000          0.000518    0.0652    0.321   86.026   
4          0.000765          0.001020    0.3640    0.521   99.994   
...             ...               ...       ...      ...      ...   
21109      0.257000          0.000000    0.0984    0.874  129.918   
21110      0.316000          0.000000    0.0623    0.495  109.079   
21111      0.075100          0.000000    0.0879    0.563   60.031   
21112      0.155000          0.000000    0.0544    0.750  119.031   
21113      0.392000          0.000000    0.0704    0.455  103.035   

       duration_ms  time_signature  
0           234507               4  
1           185334               4  
2           216000               4  
3           226640               4  
4           246467               4  
...            ...             ...  
21109       188760               4  
21110       173413               4  
21111       166280               4  
21112       153533               4  
21113       177653               4  

[21114 rows x 17 columns]

Removing columns which will not help me building my machine learning algorythms.

In [137]:
df2.to_csv(r'\Users\Dubun\IronHack\Other\Labs\Final project\database\df_top10 v271021.csv', index=False, header = True)

In [20]:
cd C:\Users\Dubun\IronHack\Other\Labs\Final project\database\

C:\Users\Dubun\IronHack\Other\Labs\Final project\database


In [21]:
df = pd.read_csv('df_top10 v271021.csv')
df

track_id               artist_id        artist_name  \
0      3RvVFTMSlEXJjtoLrcDflF  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
1      5w8bF2RGgf9PkXEoEl8SSN  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
2      5u9zU1N1urKWaULrbEhMpr  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
3      1TC9cYi9ZlEAhpoyd5sYn4  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
4      2YiGsf5CLUtgty23h1RNOQ  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
...                       ...                     ...                ...   
21109  4kP1f09mjeK56QEykWC7Sk  2y7twK2fqUipb5Ef2hNrdO                KIZ   
21110  0LA1EZMkLVyafay8DQjS21  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21111  6JYM6Dz5szNhAIHlAd37Cb  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21112  6frPNZ8woC8jH3rgjRRSUL  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21113  7xR1eN0b55dXhf4eyIXTZl  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   

      artist_genre  danceability  energy  key  loudness  mode  speechiness  \
0             rock         0.499   0.362    7   -16.331     1       0.0378   
1             rock         0.398   0.685    7    -9.098     0       0.0417   
2             rock         0.714   0.928    0    -7.118     1       0.0385   
3             rock         0.519   0.728    4    -6.660     0       0.0290   
4             rock         0.574   0.635    3    -6.683     1       0.0242   
...            ...           ...     ...  ...       ...   ...          ...   
21109          pop         0.762   0.451    5    -9.501     0       0.0848   
21110          pop         0.709   0.724    3    -4.928     1       0.0608   
21111          pop         0.429   0.787    5    -6.510     0       0.2840   
21112          pop         0.724   0.686    8    -5.040     0       0.0660   
21113          pop         0.620   0.666    1    -5.355     0       0.0762   

       acousticness  instrumentalness  liveness  valence    tempo  \
0          0.186000          0.175000    0.1190    0.354  177.691   
1          0.417000          0.005580    0.0787    0.376  109.994   
2          0.034300          0.746000    0.1370    0.487  120.007   
3          0.103000          0.000518    0.0652    0.321   86.026   
4          0.000765          0.001020    0.3640    0.521   99.994   
...             ...               ...       ...      ...      ...   
21109      0.257000          0.000000    0.0984    0.874  129.918   
21110      0.316000          0.000000    0.0623    0.495  109.079   
21111      0.075100          0.000000    0.0879    0.563   60.031   
21112      0.155000          0.000000    0.0544    0.750  119.031   
21113      0.392000          0.000000    0.0704    0.455  103.035   

       duration_ms  time_signature  
0           234507               4  
1           185334               4  
2           216000               4  
3           226640               4  
4           246467               4  
...            ...             ...  
21109       188760               4  
21110       173413               4  
21111       166280               4  
21112       153533               4  
21113       177653               4  

[21114 rows x 17 columns]

# SPOTIFY TO GET TRACK POPULARITY, NB OF FOLLOWERS & RELEASE DATE

In [184]:
AUTH_URL = 'https://accounts.spotify.com/api/token'

# POST
auth_response = requests.post(AUTH_URL, {
    'grant_type': 'client_credentials',
    'client_id': client_id,
    'client_secret': client_secret,
})

# convert the response to JSON
auth_response_data = auth_response.json()

# save the access token
access_token = auth_response_data['access_token']

headers = {
    'Authorization': 'Bearer {token}'.format(token=access_token)
}

## TRACK POPULARITY

In [265]:
BASE_URL = 'https://api.spotify.com/v1/tracks'
track_popularity = []
track_ids=[]

for j in range(21056, 21116, 60): #looping through track_id
    track_sequence = track_id[j:j+60] #by list of 500
    for i in range(j, j+60): 
        tracks=track_id[i]
        fetch = requests.get(BASE_URL + '/' + tracks, headers=headers).json()
        try:
            track_ids.append(track_id[i])
        except:
            pass
        try:
            if 'popularity' in fetch.keys():
                track_popularity.append(fetch['popularity'])
            else:
                track_popularity.append(None)
                print("warning, should reset")
        except:
            pass
    start = start + 60
    sleep(randint(91,121))
    popularity = pd.DataFrame.from_dict({'track_id': track_ids, 'popularity': track_popularity}, orient = 'index').transpose()
    popularity.to_csv(r'C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_popularity\21056_{0}.csv'.format(j+60), index=False, header = True)
    print("saved {0} tracks so far".format(j+60))

saved 21116 tracks so far


Retrieving track popularity from spotify API. This will be my target variable.

In [5]:
 cd C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_popularity

C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_popularity


In [7]:
track_popularity = pd.DataFrame([])

for file_name in glob.glob(r'C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_popularity\*.csv'):
    df6 = pd.read_csv(file_name)
    track_popularity = pd.concat([track_popularity, df6])
track_popularity = track_popularity.drop_duplicates(subset=['track_id'])
track_popularity

track_id  popularity
0    4rIC8HTqJdek2heOWzaqPA          48
1    223woKGpVuDG0hiK3DWU2H          46
2    3sNtljPdu1zkGeSLWI1bTL          45
3    4CuiupyPJDaVArnqDO7hh0          44
4    4Or4rIECXvExLB1ZimzFmj          36
..                      ...         ...
191  2OQ6uGoUz5Op50QNQ8ffsR          31
194  0L0OD7FJQlNKkJtlfFEs5o          29
195  51YwclZzTNvyjY1pFONHMV          29
197  3CXTzLgriwhrJC098rh5C4          25
199  5P4k002tsyuXZpHTpKyifG          52

[20589 rows x 2 columns]

In [8]:
track_popularity.to_csv(r'C:\Users\Dubun\IronHack\Other\Labs\Final project\database\main_df.csv')

In [10]:
cd C:\Users\Dubun\IronHack\Other\Labs\Final project\database

C:\Users\Dubun\IronHack\Other\Labs\Final project\database


In [4]:
track_popularity = pd.read_csv(r'C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_popularity\main_df.csv')
track_popularity

C:\ProgramData\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3146: DtypeWarning: Columns (4,5,6) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


Unnamed: 0                track_id  popularity  Unnamed: 0.1  \
0                 0  4rIC8HTqJdek2heOWzaqPA          48           NaN   
1                 1  223woKGpVuDG0hiK3DWU2H          46           NaN   
2                 2  3sNtljPdu1zkGeSLWI1bTL          45           NaN   
3                 3  4CuiupyPJDaVArnqDO7hh0          44           NaN   
4                 4  4Or4rIECXvExLB1ZimzFmj          36           NaN   
...             ...                     ...         ...           ...   
1029523      679309  4LgWFP2niEy4DQuG61fzLR           4      329095.0   
1029524      679310  1X90OmDuwYZyzAhY2cnp3R           4      329096.0   
1029525      679311  1buJZY15IGt89a7tELTMd7           4      329097.0   
1029526      679312  2eDvflMCe9ssOkUJrxqikB          48      329098.0   
1029527      679313  126HWk9JuCt4PHZJAAsgOv          46      329099.0   

        artist_id artist_name artist_genre  danceability  energy  key  ...  \
0             NaN         NaN          NaN           NaN     NaN  NaN  ...   
1             NaN         NaN          NaN           NaN     NaN  NaN  ...   
2             NaN         NaN          NaN           NaN     NaN  NaN  ...   
3             NaN         NaN          NaN           NaN     NaN  NaN  ...   
4             NaN         NaN          NaN           NaN     NaN  NaN  ...   
...           ...         ...          ...           ...     ...  ...  ...   
1029523       NaN         NaN          NaN           NaN     NaN  NaN  ...   
1029524       NaN         NaN          NaN           NaN     NaN  NaN  ...   
1029525       NaN         NaN          NaN           NaN     NaN  NaN  ...   
1029526       NaN         NaN          NaN           NaN     NaN  NaN  ...   
1029527       NaN         NaN          NaN           NaN     NaN  NaN  ...   

         speechiness  acousticness  instrumentalness  liveness  valence  \
0                NaN           NaN               NaN       NaN      NaN   
1                NaN           NaN               NaN       NaN      NaN   
2                NaN           NaN               NaN       NaN      NaN   
3                NaN           NaN               NaN       NaN      NaN   
4                NaN           NaN               NaN       NaN      NaN   
...              ...           ...               ...       ...      ...   
1029523          NaN           NaN               NaN       NaN      NaN   
1029524          NaN           NaN               NaN       NaN      NaN   
1029525          NaN           NaN               NaN       NaN      NaN   
1029526          NaN           NaN               NaN       NaN      NaN   
1029527          NaN           NaN               NaN       NaN      NaN   

         tempo  duration_ms  time_signature  Unnamed: 0.1.1  Unnamed: 0.1.1.1  
0          NaN          NaN             NaN             NaN               NaN  
1          NaN          NaN             NaN             NaN               NaN  
2          NaN          NaN             NaN             NaN               NaN  
3          NaN          NaN             NaN             NaN               NaN  
4          NaN          NaN             NaN             NaN               NaN  
...        ...          ...             ...             ...               ...  
1029523    NaN          NaN             NaN           145.0               NaN  
1029524    NaN          NaN             NaN           146.0               NaN  
1029525    NaN          NaN             NaN           147.0               NaN  
1029526    NaN          NaN             NaN           148.0               NaN  
1029527    NaN          NaN             NaN           149.0               NaN  

[1029528 rows x 22 columns]

In [26]:
track_popularity = track_popularity.drop_duplicates(subset=['track_id'])
track_popularity

Unnamed: 0                track_id  popularity  Unnamed: 0.1  \
0                0  4rIC8HTqJdek2heOWzaqPA          48           NaN   
1                1  223woKGpVuDG0hiK3DWU2H          46           NaN   
2                2  3sNtljPdu1zkGeSLWI1bTL          45           NaN   
3                3  4CuiupyPJDaVArnqDO7hh0          44           NaN   
4                4  4Or4rIECXvExLB1ZimzFmj          36           NaN   
...            ...                     ...         ...           ...   
328791         191  2OQ6uGoUz5Op50QNQ8ffsR          31           NaN   
328794         194  0L0OD7FJQlNKkJtlfFEs5o          29           NaN   
328795         195  51YwclZzTNvyjY1pFONHMV          29           NaN   
328797         197  3CXTzLgriwhrJC098rh5C4          25           NaN   
328799         199  5P4k002tsyuXZpHTpKyifG          52           NaN   

       artist_id artist_name artist_genre  danceability  energy  key  ...  \
0            NaN         NaN          NaN           NaN     NaN  NaN  ...   
1            NaN         NaN          NaN           NaN     NaN  NaN  ...   
2            NaN         NaN          NaN           NaN     NaN  NaN  ...   
3            NaN         NaN          NaN           NaN     NaN  NaN  ...   
4            NaN         NaN          NaN           NaN     NaN  NaN  ...   
...          ...         ...          ...           ...     ...  ...  ...   
328791       NaN         NaN          NaN           NaN     NaN  NaN  ...   
328794       NaN         NaN          NaN           NaN     NaN  NaN  ...   
328795       NaN         NaN          NaN           NaN     NaN  NaN  ...   
328797       NaN         NaN          NaN           NaN     NaN  NaN  ...   
328799       NaN         NaN          NaN           NaN     NaN  NaN  ...   

        speechiness  acousticness  instrumentalness  liveness  valence  tempo  \
0               NaN           NaN               NaN       NaN      NaN    NaN   
1               NaN           NaN               NaN       NaN      NaN    NaN   
2               NaN           NaN               NaN       NaN      NaN    NaN   
3               NaN           NaN               NaN       NaN      NaN    NaN   
4               NaN           NaN               NaN       NaN      NaN    NaN   
...             ...           ...               ...       ...      ...    ...   
328791          NaN           NaN               NaN       NaN      NaN    NaN   
328794          NaN           NaN               NaN       NaN      NaN    NaN   
328795          NaN           NaN               NaN       NaN      NaN    NaN   
328797          NaN           NaN               NaN       NaN      NaN    NaN   
328799          NaN           NaN               NaN       NaN      NaN    NaN   

        duration_ms  time_signature  Unnamed: 0.1.1  Unnamed: 0.1.1.1  
0               NaN             NaN             NaN               NaN  
1               NaN             NaN             NaN               NaN  
2               NaN             NaN             NaN               NaN  
3               NaN             NaN             NaN               NaN  
4               NaN             NaN             NaN               NaN  
...             ...             ...             ...               ...  
328791          NaN             NaN             NaN               NaN  
328794          NaN             NaN             NaN               NaN  
328795          NaN             NaN             NaN               NaN  
328797          NaN             NaN             NaN               NaN  
328799          NaN             NaN             NaN               NaN  

[20589 rows x 22 columns]

In [27]:
df_with_t_pop = df.merge(track_popularity, left_on = 'track_id', right_on = 'track_id')
df_with_t_pop

track_id             artist_id_x      artist_name_x  \
0      3RvVFTMSlEXJjtoLrcDflF  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
1      5w8bF2RGgf9PkXEoEl8SSN  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
2      5u9zU1N1urKWaULrbEhMpr  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
3      1TC9cYi9ZlEAhpoyd5sYn4  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
4      2YiGsf5CLUtgty23h1RNOQ  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
...                       ...                     ...                ...   
21109  4kP1f09mjeK56QEykWC7Sk  2y7twK2fqUipb5Ef2hNrdO                KIZ   
21110  0LA1EZMkLVyafay8DQjS21  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21111  6JYM6Dz5szNhAIHlAd37Cb  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21112  6frPNZ8woC8jH3rgjRRSUL  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21113  7xR1eN0b55dXhf4eyIXTZl  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   

      artist_genre_x  danceability_x  energy_x  key_x  loudness_x  mode_x  \
0               rock           0.499     0.362      7     -16.331       1   
1               rock           0.398     0.685      7      -9.098       0   
2               rock           0.714     0.928      0      -7.118       1   
3               rock           0.519     0.728      4      -6.660       0   
4               rock           0.574     0.635      3      -6.683       1   
...              ...             ...       ...    ...         ...     ...   
21109            pop           0.762     0.451      5      -9.501       0   
21110            pop           0.709     0.724      3      -4.928       1   
21111            pop           0.429     0.787      5      -6.510       0   
21112            pop           0.724     0.686      8      -5.040       0   
21113            pop           0.620     0.666      1      -5.355       0   

       speechiness_x  ...  speechiness_y  acousticness_y  instrumentalness_y  \
0             0.0378  ...            NaN             NaN                 NaN   
1             0.0417  ...            NaN             NaN                 NaN   
2             0.0385  ...            NaN             NaN                 NaN   
3             0.0290  ...            NaN             NaN                 NaN   
4             0.0242  ...            NaN             NaN                 NaN   
...              ...  ...            ...             ...                 ...   
21109         0.0848  ...            NaN             NaN                 NaN   
21110         0.0608  ...            NaN             NaN                 NaN   
21111         0.2840  ...            NaN             NaN                 NaN   
21112         0.0660  ...            NaN             NaN                 NaN   
21113         0.0762  ...            NaN             NaN                 NaN   

       liveness_y  valence_y  tempo_y  duration_ms_y  time_signature_y  \
0             NaN        NaN      NaN            NaN               NaN   
1             NaN        NaN      NaN            NaN               NaN   
2             NaN        NaN      NaN            NaN               NaN   
3             NaN        NaN      NaN            NaN               NaN   
4             NaN        NaN      NaN            NaN               NaN   
...           ...        ...      ...            ...               ...   
21109         NaN        NaN      NaN            NaN               NaN   
21110         NaN        NaN      NaN            NaN               NaN   
21111         NaN        NaN      NaN            NaN               NaN   
21112         NaN        NaN      NaN            NaN               NaN   
21113         NaN        NaN      NaN            NaN               NaN   

       Unnamed: 0.1.1  Unnamed: 0.1.1.1  
0                 NaN               NaN  
1                 NaN               NaN  
2                 NaN               NaN  
3                 NaN               NaN  
4                 NaN               NaN  
...               ...               ...  
21109             NaN               NaN

In [28]:
df_with_t_pop.to_csv(r'C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_popularity\df_final.csv')

In [19]:
cd C:\Users\Dubun\IronHack\Other\Labs\Final project\database

C:\Users\Dubun\IronHack\Other\Labs\Final project\database


In [20]:
df_final = pd.read_csv('df_final.csv')
df_final

track_id               artist_id        artist_name  \
0      3RvVFTMSlEXJjtoLrcDflF  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
1      5w8bF2RGgf9PkXEoEl8SSN  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
2      5u9zU1N1urKWaULrbEhMpr  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
3      1TC9cYi9ZlEAhpoyd5sYn4  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
4      2YiGsf5CLUtgty23h1RNOQ  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
...                       ...                     ...                ...   
21109  4kP1f09mjeK56QEykWC7Sk  2y7twK2fqUipb5Ef2hNrdO                KIZ   
21110  0LA1EZMkLVyafay8DQjS21  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21111  6JYM6Dz5szNhAIHlAd37Cb  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21112  6frPNZ8woC8jH3rgjRRSUL  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21113  7xR1eN0b55dXhf4eyIXTZl  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   

      artist_genre  danceability  energy  key  loudness  mode  speechiness  \
0             rock         0.499   0.362    7   -16.331     1       0.0378   
1             rock         0.398   0.685    7    -9.098     0       0.0417   
2             rock         0.714   0.928    0    -7.118     1       0.0385   
3             rock         0.519   0.728    4    -6.660     0       0.0290   
4             rock         0.574   0.635    3    -6.683     1       0.0242   
...            ...           ...     ...  ...       ...   ...          ...   
21109          pop         0.762   0.451    5    -9.501     0       0.0848   
21110          pop         0.709   0.724    3    -4.928     1       0.0608   
21111          pop         0.429   0.787    5    -6.510     0       0.2840   
21112          pop         0.724   0.686    8    -5.040     0       0.0660   
21113          pop         0.620   0.666    1    -5.355     0       0.0762   

       acousticness  instrumentalness  liveness  valence    tempo  \
0          0.186000          0.175000    0.1190    0.354  177.691   
1          0.417000          0.005580    0.0787    0.376  109.994   
2          0.034300          0.746000    0.1370    0.487  120.007   
3          0.103000          0.000518    0.0652    0.321   86.026   
4          0.000765          0.001020    0.3640    0.521   99.994   
...             ...               ...       ...      ...      ...   
21109      0.257000          0.000000    0.0984    0.874  129.918   
21110      0.316000          0.000000    0.0623    0.495  109.079   
21111      0.075100          0.000000    0.0879    0.563   60.031   
21112      0.155000          0.000000    0.0544    0.750  119.031   
21113      0.392000          0.000000    0.0704    0.455  103.035   

       duration_ms  time_signature  popularity  
0           234507               4          38  
1           185334               4          32  
2           216000               4          29  
3           226640               4          26  
4           246467               4          10  
...            ...             ...         ...  
21109       188760               4           6  
21110       173413               4          24  
21111       166280               4          22  
21112       153533               4          15  
21113       177653               4          13  

[21114 rows x 18 columns]

## NB OF FOLLOWERS PER ARTIST

In [50]:
unique_artist_ids_list = pd.unique(df_final['artist_id'])
unique_artist_ids = pd.DataFrame(unique_artist_ids_list)
unique_artist_ids

0
0     41EfC0kQmOWfbW2FBGttze
1     5XrlxolRDQ4RLX75nkVFEk
2     0qG2Tf6fJaCuWKGz1IitKa
3     4cosRYKPuU1M79bSbZcK3D
4     5VTWoYYizcOY3uIKnxeCGI
...                      ...
2231  3A08xhQd902eJmnI5U0bJe
2232  48TEPWuSU7KRr5sFpafo5q
2233  3bF6ab2QpnAww6OZ4qZM8E
2234  2y7twK2fqUipb5Ef2hNrdO
2235  5LFX3E1nLxOMLWOHSFwbG0

[2236 rows x 1 columns]

In [68]:
len(unique_artist_ids_list)

2236

In [102]:
cd C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_followers

C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_followers


In [105]:
artist_ids = []
artist_followers = []
BASE_URL = 'https://api.spotify.com/v1/artists'
for j in range(2200, 2236, 36):
    artist_sequence = unique_artist_ids_list[j:j+36]
    for i in range(j, j+36):
        artists = unique_artist_ids_list[i]
        fetch = requests.get(BASE_URL + '/' + artists, headers=headers).json()
        try:
            artist_ids.append(unique_artist_ids_list[i])
        except:
            pass
        try:
            if 'total' in fetch['followers'].keys():
                artist_followers.append(fetch['followers']['total'])
            else: 
                artist_followers.append(None)
                print("warning, should reset")
        except:
            pass
    sleep(randint(51,71))
    followers = pd.DataFrame.from_dict({'artist_id': artist_ids, 'artist_followers': artist_followers}, orient = 'index').transpose()
    followers.to_csv(r'C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_followers\2200_{0}.csv'.format(j+36), index=False, header = True)
    print("saved {0} artists so far".format(j+36))

saved 2236 artists so far


In [106]:
cd C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_followers

C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_followers


In [107]:
artists_followers = pd.DataFrame([])

for file_name in glob.glob(r'C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_followers\*.csv'):
    df7 = pd.read_csv(file_name)
    artists_followers = pd.concat([artists_followers, df7])
artists_followers = artists_followers.drop_duplicates(subset=['artist_id'])
artists_followers


artist_id  artist_followers
0    41EfC0kQmOWfbW2FBGttze             10948
1    5XrlxolRDQ4RLX75nkVFEk               143
2    0qG2Tf6fJaCuWKGz1IitKa            130721
3    4cosRYKPuU1M79bSbZcK3D               213
4    5VTWoYYizcOY3uIKnxeCGI            310373
..                      ...               ...
795  5fLCrVcmUxZL5YMdxyLK0E                 0
796  2lPL3aVXBS4gVGymFOAYxF            112560
797  5xMkiuMCSS13z5shShg7Mz             57206
798  3U0pUvFUMv6gJiTFnXiueY            228098
799  7rTPCNINGzMp3Hc8Xqht5Z            527108

[2236 rows x 2 columns]

In [108]:
artists_followers.to_csv(r'C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_followers\summary.csv')

In [109]:
df_with_followers = df_final.merge(artists_followers, left_on = 'artist_id', right_on = 'artist_id')
df_with_followers

track_id               artist_id        artist_name  \
0      3RvVFTMSlEXJjtoLrcDflF  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
1      5w8bF2RGgf9PkXEoEl8SSN  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
2      5u9zU1N1urKWaULrbEhMpr  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
3      1TC9cYi9ZlEAhpoyd5sYn4  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
4      2YiGsf5CLUtgty23h1RNOQ  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
...                       ...                     ...                ...   
21109  4kP1f09mjeK56QEykWC7Sk  2y7twK2fqUipb5Ef2hNrdO                KIZ   
21110  0LA1EZMkLVyafay8DQjS21  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21111  6JYM6Dz5szNhAIHlAd37Cb  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21112  6frPNZ8woC8jH3rgjRRSUL  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21113  7xR1eN0b55dXhf4eyIXTZl  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   

      artist_genre  danceability  energy  key  loudness  mode  speechiness  \
0             rock         0.499   0.362    7   -16.331     1       0.0378   
1             rock         0.398   0.685    7    -9.098     0       0.0417   
2             rock         0.714   0.928    0    -7.118     1       0.0385   
3             rock         0.519   0.728    4    -6.660     0       0.0290   
4             rock         0.574   0.635    3    -6.683     1       0.0242   
...            ...           ...     ...  ...       ...   ...          ...   
21109          pop         0.762   0.451    5    -9.501     0       0.0848   
21110          pop         0.709   0.724    3    -4.928     1       0.0608   
21111          pop         0.429   0.787    5    -6.510     0       0.2840   
21112          pop         0.724   0.686    8    -5.040     0       0.0660   
21113          pop         0.620   0.666    1    -5.355     0       0.0762   

       acousticness  instrumentalness  liveness  valence    tempo  \
0          0.186000          0.175000    0.1190    0.354  177.691   
1          0.417000          0.005580    0.0787    0.376  109.994   
2          0.034300          0.746000    0.1370    0.487  120.007   
3          0.103000          0.000518    0.0652    0.321   86.026   
4          0.000765          0.001020    0.3640    0.521   99.994   
...             ...               ...       ...      ...      ...   
21109      0.257000          0.000000    0.0984    0.874  129.918   
21110      0.316000          0.000000    0.0623    0.495  109.079   
21111      0.075100          0.000000    0.0879    0.563   60.031   
21112      0.155000          0.000000    0.0544    0.750  119.031   
21113      0.392000          0.000000    0.0704    0.455  103.035   

       duration_ms  time_signature  popularity  artist_followers  
0           234507               4          38             10948  
1           185334               4          32             10948  
2           216000               4          29             10948  
3           226640               4          26             10948  
4           246467               4          10             10948  
...            ...             ...         ...               ...  
21109       188760               4           6              2788  
21110       173413               4          24               990  
21111       166280               4          22               990  
21112       153533               4          15               990  
21113       177653               4          13               990  

[21114 rows x 19 columns]

In [110]:
df_with_followers.to_csv(r'C:\Users\Dubun\IronHack\Other\Labs\Final project\database\df_final2.csv')

In [111]:
cd C:\Users\Dubun\IronHack\Other\Labs\Final project\database

C:\Users\Dubun\IronHack\Other\Labs\Final project\database


In [123]:
df_final2 = pd.read_csv('df_final2.csv')
df = df_final2
df

Unnamed: 0                track_id               artist_id  \
0               0  3RvVFTMSlEXJjtoLrcDflF  41EfC0kQmOWfbW2FBGttze   
1               1  5w8bF2RGgf9PkXEoEl8SSN  41EfC0kQmOWfbW2FBGttze   
2               2  5u9zU1N1urKWaULrbEhMpr  41EfC0kQmOWfbW2FBGttze   
3               3  1TC9cYi9ZlEAhpoyd5sYn4  41EfC0kQmOWfbW2FBGttze   
4               4  2YiGsf5CLUtgty23h1RNOQ  41EfC0kQmOWfbW2FBGttze   
...           ...                     ...                     ...   
21109       21109  4kP1f09mjeK56QEykWC7Sk  2y7twK2fqUipb5Ef2hNrdO   
21110       21110  0LA1EZMkLVyafay8DQjS21  5LFX3E1nLxOMLWOHSFwbG0   
21111       21111  6JYM6Dz5szNhAIHlAd37Cb  5LFX3E1nLxOMLWOHSFwbG0   
21112       21112  6frPNZ8woC8jH3rgjRRSUL  5LFX3E1nLxOMLWOHSFwbG0   
21113       21113  7xR1eN0b55dXhf4eyIXTZl  5LFX3E1nLxOMLWOHSFwbG0   

             artist_name artist_genre  danceability  energy  key  loudness  \
0      La Maison Tellier         rock         0.499   0.362    7   -16.331   
1      La Maison Tellier         rock         0.398   0.685    7    -9.098   
2      La Maison Tellier         rock         0.714   0.928    0    -7.118   
3      La Maison Tellier         rock         0.519   0.728    4    -6.660   
4      La Maison Tellier         rock         0.574   0.635    3    -6.683   
...                  ...          ...           ...     ...  ...       ...   
21109                KIZ          pop         0.762   0.451    5    -9.501   
21110           CASANOVA          pop         0.709   0.724    3    -4.928   
21111           CASANOVA          pop         0.429   0.787    5    -6.510   
21112           CASANOVA          pop         0.724   0.686    8    -5.040   
21113           CASANOVA          pop         0.620   0.666    1    -5.355   

       mode  speechiness  acousticness  instrumentalness  liveness  valence  \
0         1       0.0378      0.186000          0.175000    0.1190    0.354   
1         0       0.0417      0.417000          0.005580    0.0787    0.376   
2         1       0.0385      0.034300          0.746000    0.1370    0.487   
3         0       0.0290      0.103000          0.000518    0.0652    0.321   
4         1       0.0242      0.000765          0.001020    0.3640    0.521   
...     ...          ...           ...               ...       ...      ...   
21109     0       0.0848      0.257000          0.000000    0.0984    0.874   
21110     1       0.0608      0.316000          0.000000    0.0623    0.495   
21111     0       0.2840      0.075100          0.000000    0.0879    0.563   
21112     0       0.0660      0.155000          0.000000    0.0544    0.750   
21113     0       0.0762      0.392000          0.000000    0.0704    0.455   

         tempo  duration_ms  time_signature  popularity  artist_followers  
0      177.691       234507               4          38             10948  
1      109.994       185334               4          32             10948  
2      120.007       216000               4          29             10948  
3       86.026       226640               4          26             10948  
4       99.994       246467               4          10             10948  
...        ...          ...             ...         ...               ...  
21109  129.918       188760               4           6              2788  
21110  109.079       173413               4          24               990  
21111   60.031       166280               4          22               990  
21112  119.031       153533               4          15               990  
21113  103.035       177653               4          13               990  

[21114 rows x 20 columns]

## RELEASE DATES

In [ ]:
BASE_URL = 'https://api.spotify.com/v1/tracks'
track_popularity = []
track_ids=[]

for j in range(21056, 21116, 60): #looping through track_id
    track_sequence = track_id[j:j+60] #by list of 500
    for i in range(j, j+60): 
        tracks=track_id[i]
        fetch = requests.get(BASE_URL + '/' + tracks, headers=headers).json()
        try:
            track_ids.append(track_id[i])
        except:
            pass
        try:
            if 'popularity' in fetch.keys():
                track_popularity.append(fetch['popularity'])
            else:
                track_popularity.append(None)
                print("warning, should reset")
        except:
            pass
    start = start + 60
    sleep(randint(91,121))
    popularity = pd.DataFrame.from_dict({'track_id': track_ids, 'popularity': track_popularity}, orient = 'index').transpose()
    popularity.to_csv(r'C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_popularity\21056_{0}.csv'.format(j+60), index=False, header = True)
    print("saved {0} tracks so far".format(j+60))

In [134]:
track_id = df['track_id'].tolist()
len(track_id)

21114

In [188]:
BASE_URL = 'https://api.spotify.com/v1/tracks'
release_dates = []
track_ids=[]

for j in range(21100, 21114, 14): #looping through track_id
    track_sequence = track_id[j:j+14] #by list of 500
    for i in range(j, j+14): 
        tracks=track_id[i]
        fetch = requests.get(BASE_URL + '/' + tracks, headers=headers).json()
        try:
            track_ids.append(track_id[i])
        except:
            pass
        try:
            if fetch['album']['release_date'] != None:
                release_dates.append(fetch['album']['release_date'])
            else:
                release_dates.append(None)
                print("warning, should reset")
        except:
            pass
    sleep(randint(91,121))
    dates = pd.DataFrame.from_dict({'track_id': track_ids, 'release_date': release_dates}, orient = 'index').transpose()
    dates.to_csv(r'C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_dates\21100_{0}.csv'.format(j+14), index=False, header = True)
    print("saved {0} dates so far".format(j+14))

saved 21114 dates so far


In [189]:
cd C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_dates

C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_dates


In [192]:
track_dates = pd.DataFrame([])

for file_name in glob.glob(r'C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_dates\*.csv'):
    df8 = pd.read_csv(file_name)
    track_dates = pd.concat([track_dates, df8])
track_dates = track_dates.drop_duplicates(subset=['track_id'])
track_dates

track_id release_date
0     3RvVFTMSlEXJjtoLrcDflF         2006
1     5w8bF2RGgf9PkXEoEl8SSN   2013-10-14
2     5u9zU1N1urKWaULrbEhMpr   2018-03-05
3     1TC9cYi9ZlEAhpoyd5sYn4   2016-01-29
4     2YiGsf5CLUtgty23h1RNOQ   2019-03-22
...                      ...          ...
2595  3fXDFWMG5Qb0y7orJV2Ab5   1997-07-01
2596  2VM2QP3iySiRQ4UWIaiQWL   1997-07-01
2597  3RGSV8PgHgHZXnGF0juGxE   2007-02-22
2598  0oTHCUOJxAzEelsUFdDVeT   2000-02-14
2599  4qnL8omnJMY0vKw49hhHH9   1997-07-01

[20587 rows x 2 columns]

In [193]:
track_dates.to_csv(r'C:\Users\Dubun\IronHack\Other\Labs\Final project\database\data_dates\summary.csv')

In [198]:
df_with_dates = df.merge(track_dates, left_on = 'track_id', right_on = 'track_id')
df = df_with_dates
df

Unnamed: 0                track_id               artist_id  \
0               0  3RvVFTMSlEXJjtoLrcDflF  41EfC0kQmOWfbW2FBGttze   
1               1  5w8bF2RGgf9PkXEoEl8SSN  41EfC0kQmOWfbW2FBGttze   
2               2  5u9zU1N1urKWaULrbEhMpr  41EfC0kQmOWfbW2FBGttze   
3               3  1TC9cYi9ZlEAhpoyd5sYn4  41EfC0kQmOWfbW2FBGttze   
4               4  2YiGsf5CLUtgty23h1RNOQ  41EfC0kQmOWfbW2FBGttze   
...           ...                     ...                     ...   
21109       21109  4kP1f09mjeK56QEykWC7Sk  2y7twK2fqUipb5Ef2hNrdO   
21110       21110  0LA1EZMkLVyafay8DQjS21  5LFX3E1nLxOMLWOHSFwbG0   
21111       21111  6JYM6Dz5szNhAIHlAd37Cb  5LFX3E1nLxOMLWOHSFwbG0   
21112       21112  6frPNZ8woC8jH3rgjRRSUL  5LFX3E1nLxOMLWOHSFwbG0   
21113       21113  7xR1eN0b55dXhf4eyIXTZl  5LFX3E1nLxOMLWOHSFwbG0   

             artist_name artist_genre  danceability  energy  key  loudness  \
0      La Maison Tellier         rock         0.499   0.362    7   -16.331   
1      La Maison Tellier         rock         0.398   0.685    7    -9.098   
2      La Maison Tellier         rock         0.714   0.928    0    -7.118   
3      La Maison Tellier         rock         0.519   0.728    4    -6.660   
4      La Maison Tellier         rock         0.574   0.635    3    -6.683   
...                  ...          ...           ...     ...  ...       ...   
21109                KIZ          pop         0.762   0.451    5    -9.501   
21110           CASANOVA          pop         0.709   0.724    3    -4.928   
21111           CASANOVA          pop         0.429   0.787    5    -6.510   
21112           CASANOVA          pop         0.724   0.686    8    -5.040   
21113           CASANOVA          pop         0.620   0.666    1    -5.355   

       mode  ...  valence    tempo  duration_ms  time_signature  popularity  \
0         1  ...    0.354  177.691       234507               4          38   
1         0  ...    0.376  109.994       185334               4          32   
2         1  ...    0.487  120.007       216000               4          29   
3         0  ...    0.321   86.026       226640               4          26   
4         1  ...    0.521   99.994       246467               4          10   
...     ...  ...      ...      ...          ...             ...         ...   
21109     0  ...    0.874  129.918       188760               4           6   
21110     1  ...    0.495  109.079       173413               4          24   
21111     0  ...    0.563   60.031       166280               4          22   
21112     0  ...    0.750  119.031       153533               4          15   
21113     0  ...    0.455  103.035       177653               4          13   

       artist_followers  in_spotify_fr_playlist  release_date_x  \
0                 10948                       0            2006   
1                 10948                       0      2013-10-14   
2                 10948                       0      2018-03-05   
3                 10948                       0      2016-01-29   
4                 10948                       0      2019-03-22   
...                 ...                     ...             ...   
21109              2788                       0      2020-07-03   
21110               990                       0      2020-12-11   
21111               990                       0      2019-10-24   
21112               990                       0      2021-03-19   
21113               990                       0      2020-03-02   

       release_date_y  release_date  
0                2006          2006  
1          2013-10-14    2013-10-14  
2          2018-03-05    2018-03-05  
3          2016-01-29    2016-01-29  
4          2019-03-22    2019-03-22  
...               ...           ...  
21109      2020-07-03    2020-07-03  
21110      2020-12-11    2020-12-11  
21111      2019-10-24    2019-10-24  
21112      2021-03-19    2021-03-19  
21113      2020-03-02    2020-03-02  

[21114 rows x 24 columns]

In [199]:
df.to_csv(r'C:\Users\Dubun\IronHack\Other\Labs\Final project\database\df_final.csv')

In [142]:
BASE_URL = 'https://api.spotify.com/v1/tracks'
fetch = requests.get(BASE_URL + '/' + '3RvVFTMSlEXJjtoLrcDflF', headers=headers).json()
fetch['album']['release_date']

'2006'

# FEATURE IN_SPOTIFY_PLAYLIST

## SPOTIFY FRANCE PLAYLISTS

In [113]:
def get_user_playlists(user_id, n_playlists=-1):
    playlist_names, playlist_ids = [], []
    
    # fetch first 20 playlists 
    results = sp.user_playlists(user=user_id, limit=20)
    user_playlists = results['items']
    time.sleep(2)
    
    # fetch playlist 21 - end
    for ofs in range(20,n_playlists,20):
        results = sp.user_playlists(user=user_id, limit=20, offset=ofs)
        user_playlists += results['items']
        time.sleep(2)
        
    # option to limit the no of playlists retreived
    if n_playlists > 0:
        if n_playlists < results['total']:
            user_playlists = user_playlists[:n_playlists]
        
    # extract name and id
    for pl in user_playlists:
        playlist_ids.append(pl['id'])
        playlist_names.append(pl['name'])
        playlist_ids = playlist_ids[:n_playlists]
        playlist_names = playlist_names[:n_playlists]
        
    # overview df
    user_playlists_total = pd.DataFrame({'name': playlist_names, 'ids': playlist_ids})
    return playlist_ids, user_playlists_total

This function enables to get playlists ids for a given spotify user.
Using it with spotify France to get 160 playlists built by spotify france (most popular ones)

In [114]:
spotify_france_playlists=get_user_playlists('spotify_france', n_playlists=160)
spotify_france_playlists

(['37i9dQZF1DWVuV87wUBNwc',
  '37i9dQZF1DXcSPhLAnCjoM',
  '37i9dQZF1DX1X23oiQRTB5',
  '37i9dQZF1DX4WpYT2fah9c',
  '37i9dQZF1DX5wB72P2sVsT',
  '37i9dQZF1DWWSuZL7uNdVA',
  '37i9dQZF1DX924zU1IARaD',
  '37i9dQZF1DX742okrrpwah',
  '37i9dQZF1DWWNlI0CRgWup',
  '37i9dQZF1DX4kZR8vL5oVX',
  '37i9dQZF1DWUSKuTscFFGY',
  '37i9dQZF1DWUI9wBgUr4BH',
  '37i9dQZF1DXdvyLcddcaVU',
  '37i9dQZF1DXbS5WTN5nKF7',
  '37i9dQZF1DX9LNmKG8TiZU',
  '37i9dQZF1DWYX0SFpLcPgx',
  '37i9dQZF1DWZafi8zKNyuo',
  '37i9dQZF1DWSBZhfF4ZHr8',
  '37i9dQZF1DX9h4FeDa1xsw',
  '37i9dQZF1DWTIykNHRogOx',
  '37i9dQZF1DX2clypTgBmEw',
  '37i9dQZF1DWXyp9AUgvDAD',
  '37i9dQZF1DX2vfncMeBctk',
  '37i9dQZF1DWSZsRTzm4O6a',
  '37i9dQZF1DXayPfy4Qnnnr',
  '37i9dQZF1DXd0Y4aXXQXWv',
  '37i9dQZF1DX69KJk2S04Hp',
  '37i9dQZF1DWSvv6VnIb3i0',
  '37i9dQZF1DWXLbJb1PtkXq',
  '37i9dQZF1DX0zyaFj8e28t',
  '37i9dQZF1DX7W8X7B8YNLZ',
  '37i9dQZF1DWZY8dtEUdfJ4',
  '37i9dQZF1DWZZvzq1xsEuc',
  '37i9dQZF1DWTypZHlgEy1G',
  '37i9dQZF1DX8OVycSP3AXz',
  '37i9dQZF1DX7agAhK

In [115]:
len(spotify_france_playlists[0])

160

In [116]:
ids_in_spotify_fr_playlists = []
for i in range(0, len(spotify_france_playlists[0])):
    ids_playlist_i = getTrackIDs('spotify_france', spotify_france_playlists[0][i])
    ids_in_spotify_fr_playlists.append(ids_playlist_i)
    i +=1
ids_in_spotify_fr_playlists

[['0gplL1WMoJ6iYaPgMCL0gX',
  '5fwSHlTEWpluwOM0Sxnh5k',
  '2Xr1dTzJee307rmrkt8c0g',
  '3uUuGVFu1V7jTQL60S1r8z',
  '46vfSYsDJZaNQcD4YvT8GD',
  '6Uj1ctrBOjOas8xZXGqKk4',
  '6bQfNiqyCX7UaQSvVVGo4I',
  '1OWLQ4FQrO13w6ssWgP8tk',
  '27NovPIUIRrOZoCHxABJwK',
  '6ngjb2Je6gEjjVOOQpWSWG',
  '27Z3UdqvdAhtJlWBMoDjBD',
  '3eJH2nAjvNXdmPfBkALiPZ',
  '6zSpb8dQRaw0M1dK8PBwQz',
  '5HCyWlXZPP0y6Gqq8TgA20',
  '5DtNh54TnwaJ8anX1FW1Ct',
  '1058fW9H3fZA6QjYCdOBad',
  '3FeVmId7tL5YN8B7R3imoM',
  '0iVkWge9ytV4Scgy2hQ2za',
  '1bwhDrXcAtqePp8mNJ0pTe',
  '2d1MywHy6FwKdzxFuSJnwl',
  '2gpWyfu7eZ01zzncHpxOtA',
  '7iSEs4lppLZQEPOG4e6BbF',
  '02MWAaffLxlfxAUY7c5dvx',
  '45PF1Y3RcW5MK0jxWvhc4D',
  '6PQ88X9TkUIAUIZJHW2upE',
  '7g0LY2NkJo9XLgd8Jl1gJU',
  '0bI7K9Becu2dtXK1Q3cZNB',
  '0e8nrvls4Qqv5Rfa2UhqmO',
  '0mV43B6pJWRjcM5TmzNe6d',
  '6CyIkPh5mRkJg7T5xXynQ8',
  '7HXBG0W8gFJwHUh5mVF9tf',
  '0NSLB3GCpO9SU69XH8jjiF',
  '3i4Va0SEnxGsEIULfouevb',
  '3Iba59sFmGhQ4kbCY5L8AX',
  '3vXnuFnC5RhPGwsFi0ORcI',
  '6OufwUcCqo81guU2j

In [117]:
spotify_fr_popular_track_ids = []
for sublist in ids_in_spotify_fr_playlists:
    for item in sublist:
        spotify_fr_popular_track_ids.append(item)
len(spotify_fr_popular_track_ids)

8923

Retrieving all track_ids from 160 playlists built by spotify France

In [118]:
spotify_fr_popular_track_ids

['0gplL1WMoJ6iYaPgMCL0gX',
 '5fwSHlTEWpluwOM0Sxnh5k',
 '2Xr1dTzJee307rmrkt8c0g',
 '3uUuGVFu1V7jTQL60S1r8z',
 '46vfSYsDJZaNQcD4YvT8GD',
 '6Uj1ctrBOjOas8xZXGqKk4',
 '6bQfNiqyCX7UaQSvVVGo4I',
 '1OWLQ4FQrO13w6ssWgP8tk',
 '27NovPIUIRrOZoCHxABJwK',
 '6ngjb2Je6gEjjVOOQpWSWG',
 '27Z3UdqvdAhtJlWBMoDjBD',
 '3eJH2nAjvNXdmPfBkALiPZ',
 '6zSpb8dQRaw0M1dK8PBwQz',
 '5HCyWlXZPP0y6Gqq8TgA20',
 '5DtNh54TnwaJ8anX1FW1Ct',
 '1058fW9H3fZA6QjYCdOBad',
 '3FeVmId7tL5YN8B7R3imoM',
 '0iVkWge9ytV4Scgy2hQ2za',
 '1bwhDrXcAtqePp8mNJ0pTe',
 '2d1MywHy6FwKdzxFuSJnwl',
 '2gpWyfu7eZ01zzncHpxOtA',
 '7iSEs4lppLZQEPOG4e6BbF',
 '02MWAaffLxlfxAUY7c5dvx',
 '45PF1Y3RcW5MK0jxWvhc4D',
 '6PQ88X9TkUIAUIZJHW2upE',
 '7g0LY2NkJo9XLgd8Jl1gJU',
 '0bI7K9Becu2dtXK1Q3cZNB',
 '0e8nrvls4Qqv5Rfa2UhqmO',
 '0mV43B6pJWRjcM5TmzNe6d',
 '6CyIkPh5mRkJg7T5xXynQ8',
 '7HXBG0W8gFJwHUh5mVF9tf',
 '0NSLB3GCpO9SU69XH8jjiF',
 '3i4Va0SEnxGsEIULfouevb',
 '3Iba59sFmGhQ4kbCY5L8AX',
 '3vXnuFnC5RhPGwsFi0ORcI',
 '6OufwUcCqo81guU2jAlDVP',
 '6I3mqTwhRpn34SLVafSH7G',
 

In [119]:
ids_in_df_final = df_final2['track_id'].tolist()
ids_in_df_final

['3RvVFTMSlEXJjtoLrcDflF',
 '5w8bF2RGgf9PkXEoEl8SSN',
 '5u9zU1N1urKWaULrbEhMpr',
 '1TC9cYi9ZlEAhpoyd5sYn4',
 '2YiGsf5CLUtgty23h1RNOQ',
 '7t1xsaC6y8tnTy1LDGkHAO',
 '3AJaqqDSA7z71SdV17zU9V',
 '4Ak5UWrKlJnxfzqhqWK0Gy',
 '732E0eyc8nzLl8fGoy8I2G',
 '0FzWLlPkhB7EtjECWTckvy',
 '5wzjMSVKtRKqwMPEBZmjDR',
 '3UyBGsmeRCaiTW559lhOo1',
 '3Nrq86aCcPztNSnwak0Sox',
 '0op4xMMqGUfOX0vNsOVoEe',
 '6UNTNXzFc4RAFK0hCTZSLi',
 '79kg43pahPRYNVLUl2drvG',
 '006otFA9ePtQIetlGAGns1',
 '6Zw0NBdfr1VlVJjDRFwUe1',
 '1XGAWEMIzbL8JiLBvhDhBw',
 '3g8awpRqI5Tq5TBgTYJhdk',
 '1l76n4YCRTxexGQr3iPXlI',
 '3Ug2Ikn8kwxJs7S40blPzM',
 '0rIMv5cxQERn91H6hcvc5W',
 '3U1ub9vyVsytNKYKT36J6j',
 '43O1ZkwbJkAoRhtpLwbT9h',
 '1Z0jOzmQza1qwdujDto5I8',
 '7pcDp5pFi0Cu6GaC3BTkiI',
 '1qXGkKwQNue3dOwVeXC7Y0',
 '0hOXVvsJ8gUnsYOaJvYVNx',
 '1VPRvbPanA2APqCJAZbVXT',
 '2Dn3DR9r8FyVvtqEpUxqXH',
 '7C0bAXqIm1Opqvx4QWSz9R',
 '3FFY7mirfTvzuckyVrLfiY',
 '6M2ZYgZ2cdu5oCHGI8wamV',
 '1CCKf8yGsKuVz5cgRwgShM',
 '6lbkdQ51nx1XPMGoiG3BZ6',
 '2uG7RbrJSq5bZWv7ufDyTg',
 

In [120]:
len(ids_in_df_final)

21114

In [127]:
in_spotify_fr_playlist = []
for i in ids_in_df_final:
    if i in spotify_fr_popular_track_ids:
        in_spotify_fr_playlist.append(1)
    else:
        in_spotify_fr_playlist.append(0)
in_spotify_fr_playlist

[0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 1,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 0,
 0,


In [128]:
df['in_spotify_fr_playlist'] = in_spotify_fr_playlist
df

Unnamed: 0                track_id               artist_id  \
0               0  3RvVFTMSlEXJjtoLrcDflF  41EfC0kQmOWfbW2FBGttze   
1               1  5w8bF2RGgf9PkXEoEl8SSN  41EfC0kQmOWfbW2FBGttze   
2               2  5u9zU1N1urKWaULrbEhMpr  41EfC0kQmOWfbW2FBGttze   
3               3  1TC9cYi9ZlEAhpoyd5sYn4  41EfC0kQmOWfbW2FBGttze   
4               4  2YiGsf5CLUtgty23h1RNOQ  41EfC0kQmOWfbW2FBGttze   
...           ...                     ...                     ...   
21109       21109  4kP1f09mjeK56QEykWC7Sk  2y7twK2fqUipb5Ef2hNrdO   
21110       21110  0LA1EZMkLVyafay8DQjS21  5LFX3E1nLxOMLWOHSFwbG0   
21111       21111  6JYM6Dz5szNhAIHlAd37Cb  5LFX3E1nLxOMLWOHSFwbG0   
21112       21112  6frPNZ8woC8jH3rgjRRSUL  5LFX3E1nLxOMLWOHSFwbG0   
21113       21113  7xR1eN0b55dXhf4eyIXTZl  5LFX3E1nLxOMLWOHSFwbG0   

             artist_name artist_genre  danceability  energy  key  loudness  \
0      La Maison Tellier         rock         0.499   0.362    7   -16.331   
1      La Maison Tellier         rock         0.398   0.685    7    -9.098   
2      La Maison Tellier         rock         0.714   0.928    0    -7.118   
3      La Maison Tellier         rock         0.519   0.728    4    -6.660   
4      La Maison Tellier         rock         0.574   0.635    3    -6.683   
...                  ...          ...           ...     ...  ...       ...   
21109                KIZ          pop         0.762   0.451    5    -9.501   
21110           CASANOVA          pop         0.709   0.724    3    -4.928   
21111           CASANOVA          pop         0.429   0.787    5    -6.510   
21112           CASANOVA          pop         0.724   0.686    8    -5.040   
21113           CASANOVA          pop         0.620   0.666    1    -5.355   

       mode  ...  acousticness  instrumentalness  liveness  valence    tempo  \
0         1  ...      0.186000          0.175000    0.1190    0.354  177.691   
1         0  ...      0.417000          0.005580    0.0787    0.376  109.994   
2         1  ...      0.034300          0.746000    0.1370    0.487  120.007   
3         0  ...      0.103000          0.000518    0.0652    0.321   86.026   
4         1  ...      0.000765          0.001020    0.3640    0.521   99.994   
...     ...  ...           ...               ...       ...      ...      ...   
21109     0  ...      0.257000          0.000000    0.0984    0.874  129.918   
21110     1  ...      0.316000          0.000000    0.0623    0.495  109.079   
21111     0  ...      0.075100          0.000000    0.0879    0.563   60.031   
21112     0  ...      0.155000          0.000000    0.0544    0.750  119.031   
21113     0  ...      0.392000          0.000000    0.0704    0.455  103.035   

       duration_ms  time_signature  popularity  artist_followers  \
0           234507               4          38             10948   
1           185334               4          32             10948   
2           216000               4          29             10948   
3           226640               4          26             10948   
4           246467               4          10             10948   
...            ...             ...         ...               ...   
21109       188760               4           6              2788   
21110       173413               4          24               990   
21111       166280               4          22               990   
21112       153533               4          15               990   
21113       177653               4          13               990   

       in_spotify_fr_playlist  
0                           0  
1                           0  
2                           0  
3                           0  
4                           0  
...                       ...  
21109                       0  
21110                       0  
21111                       0  
21112                       0  
21113                       0  

[21114 rows x 21 columns]

In [129]:
df['in_spotify_fr_playlist'].value_counts()

0    19975
1     1139
Name: in_spotify_fr_playlist, dtype: int64

Out of the 20k + tracks from my database, 1139 are listed in one of the Spotify homemade playlist.

In [194]:
df

Unnamed: 0                track_id               artist_id  \
0               0  3RvVFTMSlEXJjtoLrcDflF  41EfC0kQmOWfbW2FBGttze   
1               1  5w8bF2RGgf9PkXEoEl8SSN  41EfC0kQmOWfbW2FBGttze   
2               2  5u9zU1N1urKWaULrbEhMpr  41EfC0kQmOWfbW2FBGttze   
3               3  1TC9cYi9ZlEAhpoyd5sYn4  41EfC0kQmOWfbW2FBGttze   
4               4  2YiGsf5CLUtgty23h1RNOQ  41EfC0kQmOWfbW2FBGttze   
...           ...                     ...                     ...   
21109       21109  4kP1f09mjeK56QEykWC7Sk  2y7twK2fqUipb5Ef2hNrdO   
21110       21110  0LA1EZMkLVyafay8DQjS21  5LFX3E1nLxOMLWOHSFwbG0   
21111       21111  6JYM6Dz5szNhAIHlAd37Cb  5LFX3E1nLxOMLWOHSFwbG0   
21112       21112  6frPNZ8woC8jH3rgjRRSUL  5LFX3E1nLxOMLWOHSFwbG0   
21113       21113  7xR1eN0b55dXhf4eyIXTZl  5LFX3E1nLxOMLWOHSFwbG0   

             artist_name artist_genre  danceability  energy  key  loudness  \
0      La Maison Tellier         rock         0.499   0.362    7   -16.331   
1      La Maison Tellier         rock         0.398   0.685    7    -9.098   
2      La Maison Tellier         rock         0.714   0.928    0    -7.118   
3      La Maison Tellier         rock         0.519   0.728    4    -6.660   
4      La Maison Tellier         rock         0.574   0.635    3    -6.683   
...                  ...          ...           ...     ...  ...       ...   
21109                KIZ          pop         0.762   0.451    5    -9.501   
21110           CASANOVA          pop         0.709   0.724    3    -4.928   
21111           CASANOVA          pop         0.429   0.787    5    -6.510   
21112           CASANOVA          pop         0.724   0.686    8    -5.040   
21113           CASANOVA          pop         0.620   0.666    1    -5.355   

       mode  ...  acousticness  instrumentalness  liveness  valence    tempo  \
0         1  ...      0.186000          0.175000    0.1190    0.354  177.691   
1         0  ...      0.417000          0.005580    0.0787    0.376  109.994   
2         1  ...      0.034300          0.746000    0.1370    0.487  120.007   
3         0  ...      0.103000          0.000518    0.0652    0.321   86.026   
4         1  ...      0.000765          0.001020    0.3640    0.521   99.994   
...     ...  ...           ...               ...       ...      ...      ...   
21109     0  ...      0.257000          0.000000    0.0984    0.874  129.918   
21110     1  ...      0.316000          0.000000    0.0623    0.495  109.079   
21111     0  ...      0.075100          0.000000    0.0879    0.563   60.031   
21112     0  ...      0.155000          0.000000    0.0544    0.750  119.031   
21113     0  ...      0.392000          0.000000    0.0704    0.455  103.035   

       duration_ms  time_signature  popularity  artist_followers  \
0           234507               4          38             10948   
1           185334               4          32             10948   
2           216000               4          29             10948   
3           226640               4          26             10948   
4           246467               4          10             10948   
...            ...             ...         ...               ...   
21109       188760               4           6              2788   
21110       173413               4          24               990   
21111       166280               4          22               990   
21112       153533               4          15               990   
21113       177653               4          13               990   

       in_spotify_fr_playlist  
0                           0  
1                           0  
2                           0  
3                           0  
4                           0  
...                       ...  
21109                       0  
21110                       0  
21111                       0  
21112                       0  
21113                       0  

[21114 rows x 21 columns]

In [200]:
cd C:\Users\Dubun\IronHack\Other\Labs\Final project\database

C:\Users\Dubun\IronHack\Other\Labs\Final project\database


In [202]:
df = pd.read_csv('df_final.csv')
df = df.drop(['Unnamed: 21','Unnamed: 22'], axis = 1)
df

track_id               artist_id        artist_name  \
0      3RvVFTMSlEXJjtoLrcDflF  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
1      5w8bF2RGgf9PkXEoEl8SSN  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
2      5u9zU1N1urKWaULrbEhMpr  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
3      1TC9cYi9ZlEAhpoyd5sYn4  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
4      2YiGsf5CLUtgty23h1RNOQ  41EfC0kQmOWfbW2FBGttze  La Maison Tellier   
...                       ...                     ...                ...   
21109  4kP1f09mjeK56QEykWC7Sk  2y7twK2fqUipb5Ef2hNrdO                KIZ   
21110  0LA1EZMkLVyafay8DQjS21  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21111  6JYM6Dz5szNhAIHlAd37Cb  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21112  6frPNZ8woC8jH3rgjRRSUL  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   
21113  7xR1eN0b55dXhf4eyIXTZl  5LFX3E1nLxOMLWOHSFwbG0           CASANOVA   

      artist_genre  danceability  energy  key  loudness  mode  speechiness  \
0             rock         0.499   0.362    7   -16.331     1       0.0378   
1             rock         0.398   0.685    7    -9.098     0       0.0417   
2             rock         0.714   0.928    0    -7.118     1       0.0385   
3             rock         0.519   0.728    4    -6.660     0       0.0290   
4             rock         0.574   0.635    3    -6.683     1       0.0242   
...            ...           ...     ...  ...       ...   ...          ...   
21109          pop         0.762   0.451    5    -9.501     0       0.0848   
21110          pop         0.709   0.724    3    -4.928     1       0.0608   
21111          pop         0.429   0.787    5    -6.510     0       0.2840   
21112          pop         0.724   0.686    8    -5.040     0       0.0660   
21113          pop         0.620   0.666    1    -5.355     0       0.0762   

       ...  instrumentalness  liveness  valence    tempo  duration_ms  \
0      ...          0.175000    0.1190    0.354  177.691       234507   
1      ...          0.005580    0.0787    0.376  109.994       185334   
2      ...          0.746000    0.1370    0.487  120.007       216000   
3      ...          0.000518    0.0652    0.321   86.026       226640   
4      ...          0.001020    0.3640    0.521   99.994       246467   
...    ...               ...       ...      ...      ...          ...   
21109  ...          0.000000    0.0984    0.874  129.918       188760   
21110  ...          0.000000    0.0623    0.495  109.079       173413   
21111  ...          0.000000    0.0879    0.563   60.031       166280   
21112  ...          0.000000    0.0544    0.750  119.031       153533   
21113  ...          0.000000    0.0704    0.455  103.035       177653   

       time_signature  popularity  artist_followers  in_spotify_fr_playlist  \
0                   4          38             10948                       0   
1                   4          32             10948                       0   
2                   4          29             10948                       0   
3                   4          26             10948                       0   
4                   4          10             10948                       0   
...               ...         ...               ...                     ...   
21109               4           6              2788                       0   
21110               4          24               990                       0   
21111               4          22               990                       0   
21112               4          15               990                       0   
21113               4          13               990                       0   

       release_date  
0              2006  
1        10/14/2013  
2          3/5/2018  
3         1/29/2016  
4         3/22/2019  
...             ...  
21109      7/3/2020  
21110    12/11/2020  
21111    10/24/2019  
21112     3/19/2021  
21113      3/2/2020  

[21114 rows x 21 columns]